### DISCUSSION TITLE
SD and CF disk

### DISCUSSION THREAD

**Discussion by elekras** [Dietrich-L/CPM-65#2] [General]
I want to port the dos-65 to my 6502 SBC with SD and CF interface. I have 2 questions:
1. the block size for transfer data to/from floppy sytem is 128 or 256 bytes?
2. the block size for scsi is 512 bytes?
Regards
Fabio Sturman

**Comment by Dietrich-L** [comment]
The BDOS is always working with 256 byte blocks. The physical block size of the respective mass storage device is handled in the BIOS.
In the case of my Junior Computer, the floppy format is 16 sectors a 256 bytes. So the BIOS driver is pretty simple. In the case of my SCSI implementation, the physical block size is 512 bytes. So the BIOS has to handle sector buffering and blocking/deblocking.
The same is true for my experimental SD card driver not yet available on my repo. If you are interested - it needs a 8 bit port on a 6522 - then let me know.

For any other issues - just ask
Dietrich

**Reply by elekras** [reply]
Hi Dietrich,
I am trying to port my SD and CF read/save block routines to dos-65.
They need only a LBA address, buffer pointer (512 bytes) to put/get data and a R/W flag.
Reading-writing your SCSI drive is very similar to r/w SD or CF.
How do you address blocks in your drives using lba or track-sector-drive?
DMAV is the pointer to 256 bytes of data to transfer to or from disk?
SECS is the equvalent of LBA but for 256 bytes block?
Regards
Fabio Sturman

**Reply by Dietrich-L** [reply]
I recently ported CPM-65 to the Junior Computer ][ ([https://forum.classic-computing.de/forum/index.php?thread/26311-junior-computer/&postID=443814#post443890](url). 
There I use a slightly different concept due to the sheer size of the SD cards - file images of 1024 kB size. These images can be copied onto the SD cards in the root directory and are the handled by my driver using the base LBA of the image file as basis for disc image access. The images then can be filled with the CPM image file explorer CIFE [https://github.com/ProgrammingHobby/CPM_Image-File_Explorer](url) written by Uwe Merker using an appropriate diskdefs.
The port is currently in a private repo here on Github and will hopefully be published in the coming month. If you wish, i will send you an access now.

And yes, you got it right:
- DMAV is the vector for a 265 page anywhere in memory, where the BDOS places or fetches the 256 bytes sector to be read/written
- SECS is the 24 bit sector nr of a 256 byte sector on the disc with an offset of the sectors used by the operating system. This offset is fixed in the BIOS table describing the disc parameters and is usually 000032, but in the case of my SCSI implementation can be larger to allow multiple CPM-65 drives on a single SCSI drive. On hindsight, i believe my more recent image file approach is much more elegant and flexible.

Dietrich
